src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -331,6 +331,7 @@ protected override void StartResource(ODataResource resource)
                 this.ResourceType,
                 resource.Properties,
                 false /* isComplexValue */,
+                this.ShouldOmitNullValues(),

```

Comment:
why do you pass this in, rather than have the jsonLightResourceSerializer get it from the Input Context?